docs/google-one-tap.md

```diff
@@ -0,0 +1,134 @@
+# Google One Tap
+
+Allow users to directly sign in to The Guardian using their Google accounts straight from the homepage.
+
+![Google One Tap banner on www.theguardian.com](images/google-one-tap.png)
+
+# Client Side Implementation
+
+Googles recommended approach is to use their [GSI library which adds the required scripts to load Google One Tap onto a page](https://developers.google.com/identity/gsi/web/guides/get-google-api-clientid#load_the_client_library), however this creates a host of issues and concerns around privacy and performance. Googles GSI bundle is ~250KB uncompressed, and includes a lot of behaviour unrelated to Google One Tap which might track the user and require unecessary compute to parse.
+
+We decided to avoid using GSI and instead build the Google One Tap functionality ourselves, as it turns out this is fairly trivial to do. Previously Google One Tap was rendered by an IFrame which was initialized by the GSI bundle, however the past few years Browsers have been implementing the new [FedCM API](https://developer.mozilla.org/en-US/docs/Web/API/FedCM_API) which natively implements Google One Tap (and other providers). Google themselves suggest using the [FedCM API](https://developers.google.com/identity/gsi/web/guides/fedcm-migration) for new applications.
+
+To use Google One Tap with FedCM we're told to enable the [`use_fedcm_for_prompt`](https://developers.google.com/identity/gsi/web/reference/js-reference#use_fedcm_for_prompt) parameter when initializing GSI, but under the hood this just calls the native FedCM API to render the prompt, something we can do ourselves. The FedCM API is still fairly undocumented and experimental, and at time of writing I couldn't find any sources describing how you would use it to initialize Google One Tap, but its API is fairly intuitive to work out, I've documented a barebones implementation of it below:
+
+```js
+window.navigator.credentials
+	.get({
+		// There are various different credentials that can be retrieved using the Credentials Management API
+		// such as One Time Passwords and Public Keys, but in our case we want FedCM credentials which is configured
+		// via the `identity` property
+		identity: {
+			providers: [
+				// In theory the FedCM supports providers other than Google so you could have a "Apple One Tap" or a "Guardian One Tap"
+				// listed here too, but I'm not entirely sure how well this works in practice.
+				{
+					// Couldn't find this endpoint documented anywhere but discovered it
+					// by observing GSI network traffic.
+					configURL: 'https://accounts.google.com/gsi/fedcm.json',
```

Comment:
how safe is it to use this endpoint if it's not documented anywhere ? what would happen if they suddenly change it ?

Reply:
Google One Tap would stop working. However the [config.json endpoint is part of the FedCM standard](https://www.w3.org/TR/fedcm/#idp-api-config-file), so if they changed the endpoint they would still need to meet the standard. 

Im going to be adding some metrics when I do the follow up PR to set the users registration platform, so we'll be able to fire an alarm if for some reason people stop signing in / registering with Google One Tap